client/knowledge_base_search/src/containers/ResultsContainer/ResultComparison.js

```diff
@@ -32,49 +33,80 @@ class ResultComparison extends Component {
   }
 
   getFullAnswer() {
-    const { full_result_type, result, enriched_result } = this.props;
+    const { full_result_type, result, enriched_result, passages } = this.props;
 
-    return full_result_type === 'enriched' ? enriched_result : result;
+    switch(full_result_type) {
+      case 'enriched':
+        this.setState({full_answer: enriched_result});
+        break;
+      case 'passages':
+        Promise.all([
+          query('enriched', {query: "id: "+passages.document_id})
+        ]).then((doc_by_id) => {
```

Comment:
you are going to need to capture cases when the result is not there (either via error or via missing document ID) - that means handling those error cases specifically.

It also might be easier to retrieve all the full document results ahead of time then just referencing them via document ID later (instead of performing a query for every time you click "Show Full Answer")

If you do decide this is the better approach than retrieving them at the beginning, we should consider caching the results instead of performing queries on every show/hide (which will get complicated)